libminifi/src/Connection.cpp

```diff
@@ -227,8 +212,7 @@ void Connection::drain(bool delete_permanently) {
   std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
Might be not related to this PR, but are we happy just locking here? Can we not resume a blocked insertion after the queue is supposedly drained?

Reply:
I don't think we can (or want to) stop new flow files from being added to the connection queue after a `drain()`, and I would say a `put()` which resumes after it has acquired the lock happens after the `drain()`, even if it started before.

But yes, I agree it's not related to this change. :)